### PR TITLE
Add TRACE level, func, and method

### PR DESCRIPTION
**Pull request by aviflax**
@timehop/backend please :eyes:

### CODE REVIEW

File: log/logging.go [hunk 1]
```diff
@@ -250,6 +264,17 @@ func (s *Logger) Debug(description string, keysAndValues ...interface{}) {
...
 
+// Trace outputs an info message with an optional list of key/value pairs.
+//
+// If LogLevel is set below LevelTrace, calling this method will yield no
+// side effects.
+func (s *Logger) Trace(description string, keysAndValues ...interface{}) {
+	if s.Level < LevelTrace {
+		return
+	}
+	logMessage(s.l, s.ID, "TRACE", description, keysAndValues...)
```
Comment (kjsteuer) [inline log/logging.go]: :nail_care: use a constants for these names but is doesn' t looks like we are today. I am guessing this line and line 166 need to be the same string.

**Comment by kjsteuer** [comment]
:chompy"

**Comment by heewa** [comment]
What's the "TRACE" level? Where does it fall in the order of the levels? What's the use?

**Comment by brunodecarvalho** [comment]
I don't oppose this but I tend to clean up logging statements that fall under "trace" category before submitting PR's.

**Comment by aviflax** [comment]
@brunodecarvalho that’s fine, but this would mean that you wouldn’t *have* to remove them… you could leave them in there and they would normally be no-ops, and they’d be harmless and in fact they might come in handy at some future point down the line.

**Comment by aviflax** [comment]
@heewa Trace would be the new highest level, beyond Debug. Sometimes when debugging, troubleshooting, investigating, etc, devs add some crazy-voluminous logging messages. Things that would flood the log making even the debug level impossible to follow for most purposes, even for short periods of time (and some devs habitually run at the debug level locally). Today, we just have to remove them once we’re done with that session. With this additional level, we could put them in as Trace statements and just leave them in place, and they might come in handy down the line.

**Comment by heewa** [comment]
Awesome, makes sense to me.

**Comment by brunodecarvalho** [comment]
> they’d be harmless and in fact they might come in handy at some future point down the line.

I can see this but I've seen code with Trace statements in the past and it just feels cluttered and polluted to me. I chose to leave trace out when we originally wrote this.

That being said, I don't strongly oppose.

**Comment by aviflax** [comment]
Let’s give it a try and see if it sticks. We don’t have to use it if it turns out we don’t like it.

Thanks all!

**Pull request merged**